a modal window for authorization has been implemented. Implemented au…

Max2002 opened the pull request:
…thorization and a request to the api upon receipt of the user. Made validation and utilities for it.

Also made a mobile version for the header.

Max2002 committed:
a modal window for authorization has been implemented. Implemented authorization and a request to the api upon receipt of the user. Made validation and utilities for it.
Also made a mobile version for the header.

itzVa1kyrie commented on src/assets/icons/index.js:
```diff
@@ -1,4 +1,29 @@
 import { ReactComponent as LogoSvg } from './logo.svg';
```
already export instead of import

itzVa1kyrie commented on src/assets/icons/index.js:
```diff
@@ -1,4 +1,29 @@
+import { ReactComponent as HomeSvg } from './home.svg';
+import { ReactComponent as ContactsSvg } from './contacts.svg';
 
-export { LogoSvg, SignInSvg };
```
delete all export here

GCoder-13 commented on src/assets/icons/index.js:
```diff
@@ -1,4 +1,29 @@
 import { ReactComponent as LogoSvg } from './logo.svg';
 import { ReactComponent as SignInSvg } from './sigIn.svg';
+import { ReactComponent as ManIconSvg } from './manIcon.svg';
```
export { ReactComponent as ManIconSvg } from './manIcon.svg';

Max2002 committed:
delete comments in api.js and fixed file .gitignore

itzVa1kyrie commented on src/components/Input/index.js:
```diff
@@ -0,0 +1,55 @@
+  const [field, meta] = useField({ name });
+  const { touched, error } = meta;
+  const isTouched = touched && error;
+  const iconError =
```
Add break line before and do function instead of variable with name "**render**IconError"

GCoder-13 replied:
use function with prefix render

GCoder-13 commented on src/components/Input/index.js:
```diff
@@ -0,0 +1,55 @@
+      {isTouched ? (
+        <ErrorMessage name={name} className={st.error} component="div" />
+      ) : (
+        <div className={st.inputMessage}>Type any valid {name}</div>
```
delete

GCoder-13 commented on src/components/Input/index.js:
```diff
@@ -0,0 +1,55 @@
+        {iconError}
+      </div>
+      {isTouched ? (
+        <ErrorMessage name={name} className={st.error} component="div" />
```
use componet Feedback from Bootstarp

GCoder-13 commented on src/components/Input/index.js:
```diff
@@ -0,0 +1,55 @@
+  return (
+    <div className={st.wrapper}>
+      <div className={st.wrapperField}>
+        {svg}
```
not svg. It’s icon

GCoder-13 commented on src/components/Input/input.module.scss:
```diff
@@ -0,0 +1,85 @@
+  margin-left: 10px;
+}
+
+@media (max-width: 768px) {
```
use variables and mixins for media

GCoder-13 commented on src/assets/styles/general.scss:
```diff
@@ -20,8 +20,20 @@ a {
+  font-family: $fontSegoeUi;
+}
+
+@media (max-width: 768px) {
```
use variables and mixins for media

itzVa1kyrie commented on src/components/Input/input.module.scss:
```diff
@@ -0,0 +1,85 @@
+  padding: 10px 0 10px 35px;
+}
+
+input::placeholder {
```
input {
  ...
  &::placeholder
}

GCoder-13 commented on src/redux/selectors/getUserSelectors.js:
```diff
@@ -0,0 +1,18 @@
+import { createSelector } from '@reduxjs/toolkit';
```
Don’t  use @reduxjs/toolkit

GCoder-13 replied:
use reselect

itzVa1kyrie commented on src/redux/actions/index.js:
```diff
@@ -0,0 +1,3 @@
+import { GET_AUTH_USER } from './actionGetAuthUser';
```
export instead of import

GCoder-13 commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,126 @@
+import { useDispatch } from 'react-redux';
+import { validationEmail, validationPassword } from '../../utils/validation';
+import st from './signIn.module.scss';
+import Field from '../../components/Input';
```
create index file for all components

GCoder-13 commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,126 @@
+        validate={(values) => validationForm(values)}
+        onSubmit={(data) => onSubmitForm(data)}
+      >
+        {() => {
```
felete function

GCoder-13 commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,126 @@
+      <div className={st.title}>Sign In</div>
+      <Formik
+        initialValues={initialValues}
+        validate={(values) => validationForm(values)}
```
validate={validationForm}

GCoder-13 commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,126 @@
+      <Formik
+        initialValues={initialValues}
+        validate={(values) => validationForm(values)}
+        onSubmit={(data) => onSubmitForm(data)}
```
onSubmit={onSubmitForm}

GCoder-13 commented on src/sections/header/index.js:
```diff
@@ -1,34 +1,118 @@
+  const user = useSelector(welcomeUserSelector);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
```
create custom hook

GCoder-13 commented on src/utils/validation.js:
```diff
@@ -0,0 +1,7 @@
+export const validationEmail = (email) =>
+  !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(email);
```
must be valid value

GCoder-13 commented on src/utils/validation.js:
```diff
@@ -0,0 +1,7 @@
+  !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(email);
+
+export const validationPassword = (password) =>
+  !/^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/.test(
```
must be valid value

GCoder-13 commented on src/sections/header/index.js:
```diff
@@ -1,34 +1,118 @@
+    setIsAuth(false);
+  };
+
+  const authUser = () => {
```
create new component

GCoder-13 commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,126 @@
+  };
+
+  const contentInputs = () =>
+    Object.keys(inputs).map((item) => {
```
dont use map

GCoder-13 commented on src/redux/selectors/getUserSelectors.js:
```diff
@@ -0,0 +1,18 @@
+const userLastSelector = (state) => state.reducerUsers.name?.last;
+const userAvatarSelector = (state) => state.reducerUsers.picture?.large;
+
+const welcomeUserSelector = createSelector(
```
create this selector in component

Max2002 committed:
Improved api request.
Changed exports of files svg.
ErrorMessage use Feedback component with react-bootsrap.
Renamed several variables.
Changed createSelector with reduxToolkit to createSelector with  reduxThunk.
Chenged utils validation, created custom hook for receiving window width and created component for menu in header, also all header adapted for mobile.
Created actions to request the API on getting the user which control status.
Use mixins in scss.

Max2002 committed:
Deleted unnecessary block in input componentn

itzVa1kyrie commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,102 @@
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { validationEmail, validationPassword } from '../../utils/validation';
+import st from './styles.module.scss';
```
move to the bottom of imports

itzVa1kyrie commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,102 @@
+  const dispatch = useDispatch();
+  const handleVisiblePass = () => setIsVisiblePass(!isVisiblePass);
+
+  const validationForm = (values) => {
```
destruction already ({ email, password })

itzVa1kyrie commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,102 @@
+    handleSignIn();
+  };
+
+  const passIcon = isVisiblePass ? (
```
do the function instead of variable, and named like **render**PassIcon

itzVa1kyrie commented on src/sections/header/index.js:
```diff
@@ -1,34 +1,74 @@
+import SignIn from '../SignIn';
+import { LogoSvg } from '../../assets/icons';
+import { getMyProfile } from '../../redux/actionCreator/getMyProfile';
+import st from './styles.module.scss';
```
move to bottom of imports

Max2002 committed:
Fix import file in header.js and signIn.js

GCoder-13 commented on src/assets/styles/mixins.scss:
```diff
@@ -0,0 +1,59 @@
+  transform: translateY(-50%);
+}
+
+@mixin for-body-font-size ($width) {
```
delete all mixins with media

GCoder-13 commented on src/assets/styles/variables.scss:
```diff
@@ -2,11 +2,24 @@ $fontSegoeUi: "Segoe UI";
 $white: #fff;
 $black: #000;
+$mainColor: #1890ff;
+$colorForError: #ff4d4f;
```
Remove duplicate variables

GCoder-13 commented on src/components/Input/styles.module.scss:
```diff
@@ -0,0 +1,81 @@
+  padding: 10px 0 10px 35px;
+}
+
+input::placeholder {
```
&::placeholder {}

GCoder-13 commented on src/components/Input/styles.module.scss:
```diff
@@ -0,0 +1,81 @@
+  margin-left: 10px;
+}
+
+@media (max-width: 768px) {
```
create mixin for media

GCoder-13 commented on src/hooks/useWindowWidth.js:
```diff
@@ -0,0 +1,15 @@
+    window.addEventListener('resize', () => setWindowWidth(window.innerWidth));
+
+    return () => {
+      window.removeEventListener('resize', () => setWindowWidth);
```
not work

GCoder-13 commented on src/redux/actions/getMyProfile.js:
```diff
@@ -0,0 +1,3 @@
+export const MY_PROFILE_FETCHING = 'MY_PROFILE_FETCHING';
```
delete file

GCoder-13 commented on src/redux/actionCreator/getMyProfile.js:
```diff
@@ -0,0 +1,25 @@
+import apiUser from '../../api/apiUser';
+import { MY_PROFILE_FETCHING, MY_PROFILE_FETCHED, ERROR } from '../actions';
```
create all action pype in this file

GCoder-13 replied:
name with suffix FETCHING, SUCCESS, ERROR

GCoder-13 replied:
const GET_MY_PROFILE_FETCHING = "GET_MY_PROFILE/FETCHING"

GCoder-13 replied:
also use prefix GET, CREATE, DELETE, UPDATE

GCoder-13 commented on src/redux/reducers/index.js:
```diff
@@ -0,0 +1,4 @@
+import { combineReducers } from 'redux';
+import getMyProfile from './getMyProfile';
```
Name the reducer like an entity

GCoder-13 commented on src/redux/selectors/getMyProfile.js:
```diff
@@ -0,0 +1,19 @@
+const getMyProfile = (state) => state.getMyProfile;
+
+export const getStatus = (state) => getMyProfile(state).status;
+const getName = (state) => getMyProfile(state).name;
```
use createSelector

GCoder-13 commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,101 @@
+          />
+          <Field
+            name="password"
+            type={isVisiblePass ? 'text' : 'password'}
```
It must be controlled in Input component

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,77 @@
+        <img className={st.circleAvatar} src={large} />
+        <div className={st.dropDownList}>
+          <ul className={st.list}>
+            <li className={st.listItem}>
```
It must be reusable component

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,77 @@
+    );
+
+    return (
+      <div className={st.welcomeUser}>
```
It’s not Menu component

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,77 @@
+  }
+
+  return (
+    <Button type="button" className={st.signInBtn} onClick={handleSignIn}>
```
It’s not Menu component

GCoder-13 commented on src/redux/reducers/getMyProfile.js:
```diff
@@ -0,0 +1,30 @@
+    case ERROR: {
+      const { error, status } = action.payload;
+
+      return { ...state, error, status };
```
All properties must be in initialState

GCoder-13 commented on src/redux/reducers/getMyProfile.js:
```diff
@@ -0,0 +1,30 @@
+const initialState = {
+  name: {},
+  picture: {},
+  status: 'idle',
```
use loading true/false

Max2002 committed:
Improved mixins for all media.
Deleted duplicate variables in scss.
Changed the hook useWindowWidth to clear the event.
Changed redux selectors, actions and reducers.
Changed control type in component input.js.
Finally redesigned the menu component.

Max2002 committed:
Added mixins media in styles for files header, input and sigIn

Max2002 committed:
Remove style .container to file general.scss and changed main block in app.js file, also changed header's styles

Max2002 committed:
Renamed folder header and button to upper case first letter

GCoder-13 commented on src/assets/styles/mixins.scss:
```diff
@@ -0,0 +1,40 @@
+  transform: translateY(-50%);
+}
+
+@mixin media-breakpoint-down($size) {
```
use bool operators and variables 
https://sass-lang.com/documentation/operators/boolean

GCoder-13 commented on src/assets/styles/variables.scss:
```diff
@@ -2,11 +2,22 @@ $fontSegoeUi: "Segoe UI";
 
 $white: #fff;
 $black: #000;
+$mainColor: #1890ff;
```
dont use suffix and prefix color

GCoder-13 commented on src/assets/styles/variables.scss:
```diff
@@ -2,11 +2,22 @@ $fontSegoeUi: "Segoe UI";
 $black: #000;
+$mainColor: #1890ff;
+$colorError: #ff4d4f;
+$colorFOrLineBorder: #f0f0f0;
```
Don’t use for

Max2002 committed:
Deletd .idea

Max2002 committed:
Added component message.
Changed component textField and menu. Also refactor redux, moved authorization functionality from Header.js to redux and controle auth in app.js

GCoder-13 commented on src/App.js:
```diff
@@ -1,10 +1,23 @@
+
+  useEffect(() => {
+    if (authKey) {
+      dispatch(getMyProfile(localStorage.getItem('auth')));
```
use authKey

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,55 @@
+  });
+
+  return (
+    <>
```
it must be container

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,55 @@
+import st from './styles.module.scss';
+
+export default function Menu(props) {
+  const { label, svg, avatar, options, classNameList } = props;
```
After variables block must be line break

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,55 @@
+  };
+
+  const contentList = options.map((menuItem) => {
+    if (Array.isArray(menuItem)) {
```
menuItem cannot be an array

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,55 @@
+      !hide && (
+        <li key={title} className={st.listItem} onClick={onClick}>
+          {icon}
+          {link ? <a href="#">{title}</a> : <span>{title}</span>}
```
Use Link component and link property in option

GCoder-13 commented on src/components/Message/index.js:
```diff
@@ -0,0 +1,34 @@
+import { Button } from '../index';
+import st from './styles.module.scss';
+
+export default function Message(props) {
```
What is this component for?

GCoder-13 commented on src/components/TextField/index.js:
```diff
@@ -0,0 +1,69 @@
+export default function TextField(props) {
+  const { name, type, placeholder, prefix } = props;
+  const [isVisiblePass, setIsVisiblePass] = useState(false);
+  const [field, meta] = useField({ name });
```
If you only use the name, then don't pass it in the object

GCoder-13 commented on src/components/TextField/index.js:
```diff
@@ -0,0 +1,69 @@
+
+  const handleVisiblePass = () => setIsVisiblePass(!isVisiblePass);
+
+  const renderEye = isVisiblePass ? (
```
Variable with render prefix must be function

GCoder-13 commented on src/components/TextField/index.js:
```diff
@@ -0,0 +1,69 @@
+    if (type === 'password') {
+      return (
+        <div className={st.wrapperIconsPass}>
+          {renderEye}
```
dont use variable

GCoder-13 commented on src/components/TextField/index.js:
```diff
@@ -0,0 +1,69 @@
+      return (
+        <div className={st.wrapperIconsPass}>
+          {renderEye}
+          {isError && <IconErrorSvg className={st.iconError} />}
```
What happens if the type is password and an error occurs? Will there be 2 icons at once?

GCoder-13 commented on src/hooks/useWindowWidth.js:
```diff
@@ -0,0 +1,19 @@
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+
+  useEffect(() => {
+    const handleWindowWidth = () => {
```
use throttle with delay 200ms

GCoder-13 commented on src/redux/actionCreator/getMyProfile.js:
```diff
@@ -0,0 +1,37 @@
+
+    localStorage.setItem('auth', email);
+
+    dispatch(myProfileSuccess(data.results[0]));
```
use destructuring assignment for get first array element

GCoder-13 commented on src/redux/reducers/myProfile.js:
```diff
@@ -0,0 +1,44 @@
+    name: '',
+    picture: {},
+  },
+  loading: true,
```
default value must be false

GCoder-13 commented on src/redux/reducers/myProfile.js:
```diff
@@ -0,0 +1,44 @@
+        authKey: payload.email,
+        data: payload,
+        loading: false,
+      };
```
also need add reset error

GCoder-13 commented on src/redux/reducers/myProfile.js:
```diff
@@ -0,0 +1,44 @@
+const myProfile = (state = initialState, { type, payload }) => {
+  switch (type) {
+    case GET_MY_PROFILE_FETCHING: {
+      return { ...state };
```
add loading and reset error

GCoder-13 commented on src/redux/reducers/myProfile.js:
```diff
@@ -0,0 +1,44 @@
+      return { ...state, payload, loading: true };
+    }
+    case MY_PROFILE_LOG_OUT: {
+      localStorage.removeItem('auth');
```
The reducer should not create side effects

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,145 @@
+import st from './styles.module.scss';
+import { logOut } from '../../redux/actionCreator/getMyProfile';
+
+const welcomeUserSelector = createSelector(
```
you can use structured selector in component file

GCoder-13 commented on src/sections/SignIn/index.js:
```diff
@@ -0,0 +1,83 @@
+  password: '',
+};
+
+export default function SignIn({ handleSignIn }) {
```
The `on` prefix is used for props and `handle` for function definitions

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,145 @@
+      <div className={clsx(st.menu, { [st.menuIsAuth]: !user.authKey })}>
+        {user.authKey && deviceWidth > 768 && (
+          <ul className={st.menuNav}>
+            <li className={st.menuNavItem}>Home</li>
```
add Link

Max2002 committed:
Changed dispatcher argument to page updates.
Added container block for menu component.
Changed map in menu component.
Added links and routing to app.
Deleted component message and use react-toast.
Changed render icon for textField.
Added util throttle for custom hook (useDeviceWidth).
Set default value false for loading and reset error and loading.
Use createStructuredSelector instead of createSelector.

Max2002 committed:
recommit folder hooks

GCoder-13 commented on src/App.js:
```diff
@@ -1,11 +1,39 @@
+        <Route
+          index
+          element={
+            <WrapperApp>
```
WrapperApp is used for all routes. You can move it up.

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,51 @@
+
+  const renderMenuItem = (item) => {
+    const { title, icon, link, onClick, hide } = item;
+    const to = title === 'Home' ? '/' : `../${title}`;
```
Delete. Add link property in option

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,51 @@
+  return (
+    <div className={st.welcomeUser}>
+      <p className={st.label}>Hello! {label}</p>
+      {svg}
```
Bad name

GCoder-13 replied:
I doubt that there should be props

GCoder-13 commented on src/components/Menu/styles.module.scss:
```diff
@@ -0,0 +1,71 @@
+  }
+}
+
+.welcomeUser:hover .dropDownSvg {
```
scss selectors

GCoder-13 commented on src/components/Menu/styles.module.scss:
```diff
@@ -0,0 +1,71 @@
+  transition: .5s;
+}
+
+.welcomeUser:hover .dropDownList {
```
scss selectors

GCoder-13 commented on src/components/TextField/index.js:
```diff
@@ -0,0 +1,62 @@
+  const renderSufix = () => (
+    <div className={st.wrapperIconsPass}>
+      {type === 'password' &&
+        (isVisiblePass ? (
```
move to render function

GCoder-13 commented on src/components/TextField/index.js:
```diff
@@ -0,0 +1,62 @@
+
+  const handleVisiblePass = () => setIsVisiblePass(!isVisiblePass);
+
+  const renderSufix = () => (
```
move to jsx

GCoder-13 commented on src/components/TextField/index.js:
```diff
@@ -0,0 +1,62 @@
+  const handleVisiblePass = () => setIsVisiblePass(!isVisiblePass);
+
+  const renderSufix = () => (
+    <div className={st.wrapperIconsPass}>
```
change className

GCoder-13 commented on src/components/TextField/index.js:
```diff
@@ -0,0 +1,62 @@
+  return (
+    <div className={st.wrapper}>
+      <div className={st.wrapperField}>
+        {prefix}
```
add wrapper with styles for prefix

GCoder-13 commented on src/index.js:
```diff
@@ -1,11 +1,15 @@
-    <App />
-  </React.StrictMode>,
+  <Provider store={store}>
+    <React.StrictMode>
```
move strictmode to up

GCoder-13 commented on src/redux/actionCreator/getMyProfile.js:
```diff
@@ -0,0 +1,41 @@
+export const GET_MY_PROFILE_FETCHING = 'GET_MY_PROFILE/FETCHING';
+export const GET_MY_PROFILE_SUCCESS = 'GET_MY_PROFILE/SUCCESS';
+export const GET_MY_PROFILE_ERROR = 'GET_MY_PROFILE/ERROR';
+export const MY_PROFILE_LOG_OUT = 'MY_PROFILE/LOG_OUT';
```
You need to use suffix for async actions only

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,154 @@
+});
+
+export default function Header() {
+  const [isSignIn, setIsSignIn] = useState(false);
```
Don’t use state. You have data in store

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,154 @@
+  const deviceWidth = useDeviceWidth();
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const notify = () =>
```
line break

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,154 @@
+  const navigate = useNavigate();
+  const notify = () =>
+    toast.success('Successfully logged out', {
+      position: 'top-right',
```
This configuration must be in component props

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,154 @@
+    ];
+    const { fullName, picture, loading, authKey } = user;
+
+    const welcomeUser = (
```
delete variable

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,154 @@
+    notify();
+  };
+
+  const renderAuthBlock = () => {
```
create new component

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,154 @@
+            <Link to="/" className={st.menuNavItem}>
+              Home
+            </Link>
+            <Link to="../Contacts" className={st.menuNavItem}>
```
for link use lower case only

Max2002 committed:
Refactored wrapperApp in file app.js and move up StrictRoute in file index.js.
Changed several scss files.
Changed render icons in textField.js.
Changed block toast and changed component menu and render menu or signIn.
Changed name all link to lower case.
Added exception for any link which don`t exist and added page 404.

GCoder-13 commented on src/App.js:
```diff
@@ -1,11 +1,22 @@
-    <div className={st.container}>
-      <Header />
-    </div>
+    <BrowserRouter>
```
```<WrapperApp>{/* all routes */}</WrapperApp>```

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,57 @@
+      !hide && (
+        <li key={title} className={st.listItem} onClick={onClick}>
+          {icon}
+          {link ? <Link to={path}>{title}</Link> : <span>{title}</span>}
```
Why do you need link and path at the same time

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,57 @@
+Menu.propTypes = {
+  label: PropTypes.string.isRequired,
+  avatar: PropTypes.string.isRequired,
+  options: PropTypes.shape({
```
It’s array with objects

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -0,0 +1,57 @@
+import Skeleton from 'react-loading-skeleton';
+import { DropDownSvg } from '../../assets/icons';
+import st from './styles.module.scss';
+import 'react-loading-skeleton/dist/skeleton.css';
```
connect in global.scss

GCoder-13 commented on src/sections/404/index.js:
```diff
@@ -0,0 +1,19 @@
+      <div className={st.code}>404</div>
+      <div className={st.message}>Requested page not found!</div>
+      <div>
+        <Button type="button" className={st.back} onClick={() => navigate('/')}>
```
Don’t use anonymous function in jsx

GCoder-13 replied:
add file with constants for all routes

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,121 @@
+      <div className={clsx(st.menu, { [st.menuIsAuth]: !authKey })}>
+        {authKey && deviceWidth > 768 && (
+          <ul className={st.menuNav}>
+            <Link to="/" className={st.menuNavItem}>
```
use constants for routes

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,121 @@
+            <Link to="/" className={st.menuNavItem}>
+              Home
+            </Link>
+            <Link to="../contacts" className={st.menuNavItem}>
```
use constants for routes

GCoder-13 commented on src/sections/WrapperApp/index.js:
```diff
@@ -0,0 +1,29 @@
+    if (authKey) {
+      dispatch(getMyProfile(authKey));
+    }
+  }, []);
```
add authKey to dependencies

Max2002 committed:
Deleted wrapperApp and remove BrowserRouter to index.js.
Changed dependes in useEffect in file app.js.
Changed actions and reducers in redux.

Max2002 committed:
Changed routes constants and use in file header.js/app.js.
Changed options in menu link

GCoder-13 commented on src/assets/styles/general.scss:
```diff
@@ -1,4 +1,7 @@
 @import 'variables';
+@import "mixins";
```
The quotes must be the same everywhere.

GCoder-13 commented on src/components/Menu/styles.module.scss:
```diff
@@ -0,0 +1,76 @@
+    cursor: pointer;
+  }
+
+  &:hover .dropDownSvg {
```
What can be improved in this selector?

GCoder-13 commented on src/components/Menu/styles.module.scss:
```diff
@@ -0,0 +1,76 @@
+    transition: .5s;
+  }
+
+  &:hover .dropDownList {
```
What can be improved in this selector?

GCoder-13 commented on src/components/Menu/styles.module.scss:
```diff
@@ -0,0 +1,76 @@
+  }
+}
+
+.listItem:first-child {
```
What can be improved in this selector?

GCoder-13 commented on src/constans/routes/index.js:
```diff
@@ -0,0 +1,3 @@
+export const HOME = '/';
```
file: src/constans/routes.js

GCoder-13 commented on src/constans/routes/index.js:
```diff
@@ -0,0 +1,3 @@
+export const HOME = '/';
+export const PROFILE = 'profile';
```
/profile

GCoder-13 commented on src/constans/routes/index.js:
```diff
@@ -0,0 +1,3 @@
+export const HOME = '/';
+export const PROFILE = 'profile';
+export const CONTACTS = 'contacts';
```
/contacts

Max2002 committed:
Removed and renamed file with constants for routes.
Changed quotes in several files and removed hover effect on dropdown in file menu.module.scss.

Max2002 committed:
Added route for error and use this route when user not auth and want enter on page /profile

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,118 @@
+    {
+      title: 'Profile',
+      icon: <ManIconSvg />,
+      link: `../${PROFILE}`,
```
Use regular relative links. They are just variables.

Max2002 committed:
Сhanged use of constants with routes.

GCoder-13 commented on src/sections/404/index.js:
```diff
@@ -0,0 +1,21 @@
+export default function Page404() {
+  const navigate = useNavigate();
+
+  const backHome = () => navigate('/');
```
use route variable in all navigate

GCoder-13 commented on src/sections/Header/index.js:
```diff
@@ -0,0 +1,118 @@
+  const logOutUser = () => {
+    localStorage.removeItem('auth');
+    dispatch(logOut());
+    navigate('/');
```
variable

GCoder-13 commented on src/App.js:
```diff
@@ -1,11 +1,36 @@
       <Header />
+      <Routes>
+        <Route index element={<Home />} />
+        <Route path={authKey ? PROFILE : ERROR} element={<Profile />} />
```
No need to render the route if the user is not logged in

Max2002 committed:
Use constants routes in several file and changed route Profile if user auth and don`t auth

GCoder-13 commented on package.json:
```diff
@@ -7,6 +7,7 @@
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
+    "axios": "^0.27.2",
```
old version
